### DISCUSSION TITLE
Set language when using model.transcribe rather than whisper.decode

### DISCUSSION THREAD

**Discussion by Zoobdude** [openai/whisper#1397] [Q&A] [answered]
As the first 30 seconds of my audio file the auto detected language is incorrect therefore I need a way to set it manually. I understand I can achieve this if I use the more low level whisper.decode. However, my audio file is far longer than 30s. Is their a way I can either use model.transcribe and specify the language as an argument. Or can I use whisper.decode on files longer than 30s. Thanks.

**Comment by glangford** [comment, marked as answer]
> Is their a way I can either use model.transcribe and specify the language as an argument.

```python
result = model.transcribe(your_file, language="en", verbose=False) # for example
```
See also
* https://github.com/openai/whisper/discussions/745

**Reply by Zoobdude** [reply]
Thanks very much.

**Reply by chanheehi** [reply]
thanks for your comment
